corehq/form_processor/models.py

```diff
@@ -426,7 +426,7 @@ def read_content(self, stream=False):
             if not self.blob_bucket and '-' in str(self.attachment_id):
                 try:
                     # http://manage.dimagi.com/default.asp?239638
-                    blob = db.get(self.blob_id, self.blobdb_bucket(True))
+                    blob = db.get(self.blob_id, self.blobdb_bucket(remove_dashes=False))
                 except (KeyError, NotFound, BadName):
```

Comment:
Yep. I'm going to work on that migration today. It's a little messy because to migrate sharded tables I need to write SQL functions :/